CHAID/CHAID.py

```diff
@@ -42,20 +43,30 @@ def sub_non_floats(self, vect):
         ----------
         tuple : prcoessed vector, hash or substitutions
         """
-        self._arr = np.array(vect)
         if vect.dtype != float and vect.dtype != int:
-            unique_v = np.unique(self._arr)
-            float_map = [(x, float(i)) for i, x in enumerate(unique_v)]
-            for value, new_id in float_map:
-                self._arr[self._arr == value] = new_id
-            self._arr = self._arr.astype(float, subok=False, copy=False)
+            unique = np.unique(vect)
```

Comment:
yeah this is gross, let's force convert everything, initially. This will slow it down marginally for int/float numpy arrays, but this initial up front cost is acceptable